### DISCUSSION TITLE
dnscrypt-proxy not working properly on MacOS Ventura when installed through brew

### DISCUSSION THREAD

**Discussion by s4dr0t1** [DNSCrypt/dnscrypt-proxy#2403] [Potential issues]
Hi,

I'm on Ventura 13.4 on a device with Apple Silicon and I installed dnscrypt-proxy with the help of home-brew (version 4.0.19).

```
brew install dnscrypt-proxy
# Installed version is 2.1.4
```

I made the following changes to the configuration file (/opt/homebrew/etc/dnscrypt-proxy.toml)
```
listen_addresses = ['127.0.0.1:5355', '[::1]:5355']
```

Then I set the DNS settings for my Network interface by going to System Settings -> Network -> WiFi -> Details -> DNS and put 127.0.0.1:5355 there (and removed everything else)

Finally, I registered it as a service by running the following:
```
sudo brew services start dnscrypt-proxy
```

Yet, after doing all this, when I check for my current DNS servers being used by [dnsleaktest](https://dnsleaktest.com), I get the same old stuff (I used Safari)

Any sort of help would be appreciated,
Thank you

**Comment by s4dr0t1** [comment]
Update: I restored the configuration file to its default (port 53) and reflected the same in DNS settings in the Network tab, and it all started to work just fine.

**Comment by jedisct1** [comment]
> Then I set the DNS settings for my Network interface by going to System Settings -> Network -> WiFi -> Details -> DNS and put 127.0.0.1:5355 there (and removed everything else)

I don't think MacOS allows port numbers to be specified here. Such entries are very likely ignored by the system.

Just put an IP address and use the default port, 53.

**Reply by s4dr0t1** [reply]
That seems like to be the issue here. As a workaround, I ran dnscrypt-proxy on port 5355 and ran a dnsmasq resolver at port 53, and placed the address of the dnscrypt-proxy inside the dnsmasq configuration and in the network settings, just used the address `127.0.0.1`.

Everything works fine now.

```
# /opt/homebrew/etc/dnsmasq.conf
# DNSCrypt on localhost port 5355
server=127.0.0.1#5355
```
```
$ sudo lsof +c 15 -Pni UDP:5355
Password:
COMMAND         PID USER   FD   TYPE             DEVICE SIZE/OFF NODE NAME
dnscrypt-proxy 1369 root    8u  IPv4 0x4af62091a419089f      0t0  UDP 127.0.0.1:5355


$ sudo lsof +c 15 -Pni UDP:53
COMMAND  PID   USER   FD   TYPE             DEVICE SIZE/OFF NODE NAME
dnsmasq 1640 nobody    4u  IPv4 0x4af62091a41be89f      0t0  UDP 127.0.0.1:53
```
